packages/http-specs/specs/payload/pageable/main.tsp

```diff
@@ -52,4 +52,248 @@ namespace ServerDrivenPagination {
 
     @nextLink next?: url;
   };
+
+  @route("/continuationtoken")
+  namespace ContinuationToken {
+    @scenario
+    @scenarioDoc("""
+      Test case for using continuation token as pagination. Continuation token is passed in the request body and response body.
+      
+      Two requests need to be tested.
+      1. Initial request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "1", "name": "dog" },
+          { "id": "2", "name": "cat" }
+        ],
+        "nextToken": "12345"
+      }
+      ```
+      2. Next page request:
+      Expected route: /payload/pageable/server-driven-pagination/continuationtoken/request-body-response-body
+      Expected request body:
+      ```json
+      { "token": "12345" }
+      ```
+      Expected response body:
+      ```json
+      { "pets": [
+          { "id": "3", "name": "bird" },
+          { "id": "4", "name": "fish" }
+        ]
+      }
+      ```
+      """)
+    @route("/request-body-response-body")
+    @list
+    op reqBodyResBody(@continuationToken token?: string): {
```

Comment:
I didn't remember we supports `continuationToken` in request body -- and this is a GET which should not have the body.

Reply:
![image](https://github.com/user-attachments/assets/3e3efe74-c60c-4179-a4ac-16ea180aaa68)
official docs show an example so I also add similar case here.
Add @timotheeguerin for confirmation here.

Reply:
Emm, looks strange to me -- GET should not have request body.

Anyway Laurent's list does not have this https://github.com/Azure/typespec-azure/issues/2141

Reply:
Let me raise a question in typespec channel https://teams.microsoft.com/l/message/19:906c1efbbec54dc8949ac736633e6bdf@thread.skype/1739861699306?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1739861699306&teamName=Azure%20SDK&channelName=TypeSpec%20Discussion&createdTime=1739861699306

Reply:
One point. The GET here I mean you write it "get" in mockapi.

If you follow the typespec-openapi it is POST
https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvdmVyc2lvbmluZyI7CtIfcmVzdNUZaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5SZXN0O9AVSHR0cNEVVsl5OwoKQHNlcnZpY2Uoe3RpdGxlOiAiVGV4dCBhdXRob3LkAJx9KQpuYW1lc3BhY2UgxBxByBs7Cgptb2RlbCBQZXQgewogIMQoOiBzdMYdfQoKQGxpc3Qgb3AgxAhQZXRzKEBjb250aW51YXRpb25Ub2tlbiB0xAY%2FyDgpOsVLQHBhZ2VJdGVtcyBwZXRzOsRkW107xBrSQm5leHTFCslGOwp9Owo%3D&e=%40typespec%2Fopenapi3&options=%7B%7D

Reply:
Interesting. I am not sure whether TCGC is aware of this scenario.
add @tadelesh for awareness.

Let me update to POST for now.

Reply:
Debugged in TCGC, and TCGC has same behavior with openapi emitter: 
![image](https://github.com/user-attachments/assets/ba20bbea-f8bc-4a61-ac4b-68d7e2722d6d)

Reply:
Yeah, as POST at least this be valid REST (and the doc likely valid too).

Then the question is mostly on us, whether emitter supports this scenario or not.

Reply:
Will confirm with Laurent: https://github.com/Azure/typespec-azure/issues/2141#issuecomment-2665054009

Reply:
Confirmed with Laurent that continuation toke is not in scope so I will delete related scenarios.